### PR TITLE
Automate Laravel MailHog capture configuration

### LINKED ISSUE #4821: Laravel requires manually setting mail
Out of the box, Laravel requires additional configuration to work with DDEV's implementation of Mailhog.
While this is not a DDEV problem, we could automate this to improve the user experience.

### PR DESCRIPTION
**Pull request by tyler36**
## The Issue

Laravel requires additional configuration to work with DDEV's Mailhog.

With current setup, sending an email results in the following exception:

```php
   Symfony\Component\Mailer\Exception\TransportException  Connection could not be established with host "mailpit:1025": stream_socket_client(): php_network_getaddresses: getaddrinfo for mailpit failed: Temporary failure in name resolution.
```

## How This PR Solves The Issue

This PR add `MAIL_HOST=127.0.0.1` to the project's `.env`.

## Manual Testing Instructions

1. Follow quickstart guide for a Laravel project. You should see the Laravel "Welcome page"
1. Launch Mailhog to watch for email:

    ```shell
    ddev launch -m
    ```

1. Enter "tinker mode" by typing the following command:

    ```shell
    ddev artisan tinker
    ```

1. Copy and past (or type) the following command into "tinker mode". This will attempt to send an email which should be caught by Mailhog.:

    ```php
    Illuminate\Support\Facades\Mail::raw('Hello World!', function($msg) {$msg->to('example@example.com')->subject('Test Email'); });
    ```

1. Type `exit` to exit "tinker mode".

Note: if you make changes to `.env` (or any php file), you need to exit and re-enter "tinker mode" for changes to apply. 

## Related Issue Link(s)
Fixes #4821

## Release/Deployment Notes

Recent versions of DDEV automate the database settings, so it seems consistent to include the mail server.

Developers are free to opt out of DDEV managing this using currently documented best practises.

<a href="https://gitpod.io/#https://github.com/ddev/ddev/pull/4822"></a>

### CODE REVIEW

File: pkg/ddevapp/laravel.go [hunk 1]
```diff
@@ -56,6 +56,7 @@ func laravelPostStartAction(app *DdevApp) error {
 		"DB_USERNAME":   "db",
 		"DB_PASSWORD":   "db",
 		"DB_CONNECTION": dbConnection,
+		"MAIL_HOST":     "127.0.0.1",
```
Comment (rfay) [inline pkg/ddevapp/laravel.go:59]: Doesn't it need the port as well, 1025?

Comment (tyler36) [reply pkg/ddevapp/laravel.go:59]: The port is set to `1025` by default.

Composer copies over the `.env.example` which has the following mail related settings:

```env
MAIL_MAILER=smtp
MAIL_HOST=mailhog
MAIL_PORT=1025
MAIL_USERNAME=null
MAIL_PASSWORD=null
MAIL_ENCRYPTION=null
MAIL_FROM_ADDRESS="hello@example.com"
MAIL_FROM_NAME="${APP_NAME}"
```

You can see that the default mail host doesn't work OOTB.

Comment (tyler36) [reply pkg/ddevapp/laravel.go:59]: Update PR to include port if you would prefer it in

Comment (rfay) [reply pkg/ddevapp/laravel.go:59]: 1025 is an unusual default mail port (25 is normal). Oh, I see, it's already there `MAIL_PORT=1025`. But we didn't have `MAIL_HOST` set right. Thanks.

**Comment by rfay** [comment]
It doesn't look like the "send email" instructions are adequate for a novice like me to manually test. Can you add more detail about how to do it? Thanks!

**Comment by tyler36** [comment]
I updated the description to try and make it simpler.  As long as you have a working Laravel install, you should be able to enter "tinker mode" (REPL). REPL allows you to run PHP commands in a bootstraped environment. 

The command in the description tries to send an email; it it should be a simple copy/paste.

Let me know if you are getting a specific error message.

**Comment by tyler36** [comment]
I have a `ddev artisan` command in my global commands. It has the `#ddev-generated` so I assume it's in the standard install right?

**Comment by rfay** [comment]
That also means the quickstart could use it instead of `ddev exec`

**Comment by rfay** [comment]
Added 
* https://github.com/ddev/ddev/pull/4822

**Comment by rfay** [comment]
Sorry I haven't tested yet, will get to it!

**Review by rfay — APPROVED** [review]
Works perfectly, code is good, thanks so much. thanks for the good testing instructions.

**Pull request merged**